### ISSUE TITLE
WebApplicationException should be formatted

### ISSUE THREAD

**Issue by vzx** [dropwizard/dropwizard#1192]
In my eyes, the result of thrown `WebApplicationException` should be formatted like other unhandled bubbled exceptions, in `LoggingExceptionMapper`. Currently they are just passed through:

```java
        if (exception instanceof WebApplicationException) {
            return ((WebApplicationException) exception).getResponse();
        }
```

Any `WebApplicationException` which was thrown with a response entity set (eg. through `throw new WebApplicationException(Response.status(..).entity(...)...)`) will _not_ be passed to any `ExceptionMapper`: it's just processed in the way the thrower intended.

However, this leaves `WebApplicationExceptions` which were thrown with a message, but without any entity (eg. `throw new WebApplicationException("not found", 404)`). These end up in `LoggingExceptionMapper`, in the code branch shown above.

The problem is that even though a message was specified, the returned `Response` does not have any entity data. That means the error will end up in Jetty, which will just produce a generic HTML page for the associated HTTP status code, without the message.

It would be nicer to format these exceptions in the same other unhandled exceptions are formatted (like `{"code": 404, "message": "not found"}`).

Even if someone throws `WebApplicationException` with some specific format (eg. a `text/plain` entity), it will not be re-formatted, because these are not passed to the `ExceptionMapper`. And finally, if someone wants a different for _all_ `WebApplicationExceptions`, they can just implement their own `ExceptionMapper<WebApplicationException>` because the mappers are chosen by most specific first.

**Comment by vzx** [comment]
I made a pull request for this: https://github.com/dropwizard/dropwizard/pull/1193

**Comment by arteam** [comment]
Closed by #1193
